### PR TITLE
Tx approval timeout

### PR DESCRIPTION
**Pull request by Velenir**
- [x]  Transactions that aren't approved/rejected for a long time will time out
- [x] Sample implementation of `stop waiting for tx approval` user confirmation, with native `confirm`for now
- [x] Modal or toast or something for user confirmation (will be annoying with state injection in `composedProvider`)

To test start a tx,but wait a bit before confirming/rejecting in the wallet

Aims to solve the use case when some providers don't respond to `eth_sendTransaction` calls and we get infinite waiting for tx promise (spinner):

- Safe through WalletConnect app
- MetaMask mobile when rejecting tx

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1547--dexreact.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1547--dexreact.review.gnosisdev.com/storybook/)**: Component stories

File: src/api/wallet/composeProvider.ts [hunk 1]
```diff
@@ -62,6 +62,47 @@ function providerAsMiddleware(provider: Provider): JsonRpcMiddleware {
   }
 }
 
+// wait 7 sec, for now
+const DEFAULT_TX_APPROVAL_TIMEOUT = 7000
```
Comment (anxolin) [inline src/api/wallet/composeProvider.ts:66]: nice for the test, although for real, at least we need sth like 1min, right?

**Review by anxolin — COMMENTED** [review]
Great, works as expected

File: src/api/wallet/composeProvider.ts (hunk 1 above)
Comment (Velenir) [reply src/api/wallet/composeProvider.ts:66]: of course :+1:

**Comment by Velenir** [comment]
Closes #1540

**Review by anxolin — APPROVED** [review]

**Review by W3stside — APPROVED** [review]
Nice!

**Pull request merged**